libraries/WDT/src/WDT.cpp

```diff
@@ -0,0 +1,72 @@
+#include "WDT.h"
+
+// Constructor
+APM3_WDT::APM3_WDT()
+{
+    // The watchdog only runs off of the LFRC
+    am_hal_clkgen_control(AM_HAL_CLKGEN_CONTROL_LFRC_START, 0);
+    am_hal_wdt_init(&watchdogConfig);
+}
+
+// Configure the watchdog timer
+void APM3_WDT::configure(uint8_t clock, uint8_t interrupt, uint8_t reset)
+{
+  WDT->CFG_b.CLKSEL = clock; 
+  WDT->CFG_b.INTVAL = interrupt; 
+  WDT->CFG_b.RESVAL = reset; 
+}
+
+// Set the LFRC clock frequency of the watchdog timer
+void APM3_WDT::setClock(uint8_t clock)
+{
```

Comment:
Add error checking code:
`if(clock > 0x4)`
`  WDT->CFG_b.CLKSEL = clock;`
`else`
`  WDT->CFG_b.CLKSEL = AM_HAL_WDT_LFRC_CLK_DEFAULT`

Reply:
This is a good suggestion and will be implemented. Given that the library defaults to the 16HZ LFRC divider, it may be the better default option. 128 Hz has a max timeout 1.99 seconds, which is likely too quick for most users.